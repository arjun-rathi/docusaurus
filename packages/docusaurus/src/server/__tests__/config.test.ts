/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import path from 'path';
import loadConfig from '../config';

describe('loadConfig', () => {
  it('website with valid siteConfig', async () => {
    const siteDir = path.join(
      __dirname,
      '__fixtures__',
      'simple-site',
      'docusaurus.config.js',
    );
    const config = await loadConfig(siteDir);
    expect(config).toMatchSnapshot();
    expect(config).not.toEqual({});
  });

  it('website with valid config creator function', async () => {
    const siteDir = path.join(
      __dirname,
      '__fixtures__',
      'configs',
      'createConfig.config.js',
    );
    const config = await loadConfig(siteDir);
    expect(config).toMatchSnapshot();
    expect(config).not.toEqual({});
  });

  it('website with valid async config', async () => {
    const siteDir = path.join(
      __dirname,
      '__fixtures__',
      'configs',
      'configAsync.config.js',
    );
    const config = await loadConfig(siteDir);
    expect(config).toMatchSnapshot();
    expect(config).not.toEqual({});
  });

  it('website with valid async config creator function', async () => {
    const siteDir = path.join(
      __dirname,
      '__fixtures__',
      'configs',
      'createConfigAsync.config.js',
    );
    const config = await loadConfig(siteDir);
    expect(config).toMatchSnapshot();
    expect(config).not.toEqual({});
  });

  it('website with incomplete siteConfig', async () => {
    const siteDir = path.join(
      __dirname,
      '__fixtures__',
      'bad-site',
      'docusaurus.config.js',
    );
    await expect(loadConfig(siteDir)).rejects.toThrowErrorMatchingSnapshot();
  });

  it('website with useless field (wrong field) in siteConfig', async () => {
    const siteDir = path.join(
      __dirname,
      '__fixtures__',
      'wrong-site',
      'docusaurus.config.js',
    );
    await expect(loadConfig(siteDir)).rejects.toThrowErrorMatchingSnapshot();
  });

  it('website with no siteConfig', async () => {
    const siteDir = path.join(
      __dirname,
      '__fixtures__',
      'nonExisting',
      'docusaurus.config.js',
    );
    await expect(loadConfig(siteDir)).rejects.toThrowError(
      /Config file at ".*?__fixtures__[/\\]nonExisting[/\\]docusaurus.config.js" not found.$/,
    );
  });
});
