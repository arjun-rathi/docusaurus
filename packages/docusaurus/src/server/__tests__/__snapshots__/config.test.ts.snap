// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadConfig website with incomplete siteConfig 1`] = `
"\\"url\\" is required
"
`;

exports[`loadConfig website with useless field (wrong field) in siteConfig 1`] = `
"These field(s) (\\"useLessField\\",) are not recognized in docusaurus.config.js.
If you still want these fields to be in your configuration, put them in the \\"customFields\\" field.
See https://docusaurus.io/docs/docusaurus.config.js/#customfields"
`;

exports[`loadConfig website with valid async config 1`] = `
{
  "baseUrl": "/",
  "baseUrlIssueBanner": true,
  "clientModules": [],
  "customFields": {},
  "i18n": {
    "defaultLocale": "en",
    "localeConfigs": {},
    "locales": [
      "en",
    ],
  },
  "noIndex": false,
  "onBrokenLinks": "throw",
  "onBrokenMarkdownLinks": "warn",
  "onDuplicateRoutes": "warn",
  "organizationName": "endiliey",
  "plugins": [],
  "presets": [],
  "projectName": "hello",
  "scripts": [],
  "staticDirectories": [
    "static",
  ],
  "stylesheets": [],
  "tagline": "Hello World",
  "themeConfig": {},
  "themes": [],
  "title": "Hello",
  "titleDelimiter": "|",
  "url": "https://docusaurus.io",
}
`;

exports[`loadConfig website with valid async config creator function 1`] = `
{
  "baseUrl": "/",
  "baseUrlIssueBanner": true,
  "clientModules": [],
  "customFields": {},
  "i18n": {
    "defaultLocale": "en",
    "localeConfigs": {},
    "locales": [
      "en",
    ],
  },
  "noIndex": false,
  "onBrokenLinks": "throw",
  "onBrokenMarkdownLinks": "warn",
  "onDuplicateRoutes": "warn",
  "organizationName": "endiliey",
  "plugins": [],
  "presets": [],
  "projectName": "hello",
  "scripts": [],
  "staticDirectories": [
    "static",
  ],
  "stylesheets": [],
  "tagline": "Hello World",
  "themeConfig": {},
  "themes": [],
  "title": "Hello",
  "titleDelimiter": "|",
  "url": "https://docusaurus.io",
}
`;

exports[`loadConfig website with valid config creator function 1`] = `
{
  "baseUrl": "/",
  "baseUrlIssueBanner": true,
  "clientModules": [],
  "customFields": {},
  "i18n": {
    "defaultLocale": "en",
    "localeConfigs": {},
    "locales": [
      "en",
    ],
  },
  "noIndex": false,
  "onBrokenLinks": "throw",
  "onBrokenMarkdownLinks": "warn",
  "onDuplicateRoutes": "warn",
  "organizationName": "endiliey",
  "plugins": [],
  "presets": [],
  "projectName": "hello",
  "scripts": [],
  "staticDirectories": [
    "static",
  ],
  "stylesheets": [],
  "tagline": "Hello World",
  "themeConfig": {},
  "themes": [],
  "title": "Hello",
  "titleDelimiter": "|",
  "url": "https://docusaurus.io",
}
`;

exports[`loadConfig website with valid siteConfig 1`] = `
{
  "baseUrl": "/",
  "baseUrlIssueBanner": true,
  "clientModules": [],
  "customFields": {},
  "favicon": "img/docusaurus.ico",
  "i18n": {
    "defaultLocale": "en",
    "localeConfigs": {},
    "locales": [
      "en",
    ],
  },
  "noIndex": false,
  "onBrokenLinks": "throw",
  "onBrokenMarkdownLinks": "warn",
  "onDuplicateRoutes": "warn",
  "organizationName": "endiliey",
  "plugins": [
    [
      "@docusaurus/plugin-content-docs",
      {
        "path": "../docs",
      },
    ],
    "@docusaurus/plugin-content-pages",
  ],
  "presets": [],
  "projectName": "hello",
  "scripts": [],
  "staticDirectories": [
    "static",
  ],
  "stylesheets": [],
  "tagline": "Hello World",
  "themeConfig": {},
  "themes": [],
  "title": "Hello",
  "titleDelimiter": "|",
  "url": "https://docusaurus.io",
}
`;
